### PR TITLE
Auth0 create project

### PR DESCRIPTION
**Pull request by chris-hammond**
This PR addresses issue #46 . It requires users to login or create an account to access the create-project page. We should open a new issue for "attaching" project submissions to a login account. We probably need to actually display the User details in the navbar instead of simply a logout button, but I think I want to make that a separate issue.

This work is based on this blog: https://auth0.com/blog/securing-gatsby-with-auth0/

**What kind of change does this PR introduce?**
Uses Auth0 for login gating the /create-project page.

**Does this PR introduce a breaking change?**
This change *removes* the NetlifyCMS admin panel. The more I think about it, the more convinced I am that we need to "move" any content in markdown files in the repo over to Strapi. Two CMS tools in a single webapp is overkill.

**What needs to be documented once your changes are merged?**
Create new Issues:
* Display profile/avatar in the navbar (make logout a dropdown)
* Attach login profile to Project submissions
* Migrate markdown content to Strapi

### CODE REVIEW

File: src/components/ProjectTile.js [hunk 1]
```diff
@@ -4,24 +4,27 @@ import { Link } from 'gatsby'
...
-        style={{ whiteSpace: 'pre-wrap' }}
-        text={data.project_description}
-        maxLine={3}
-      />
-    </Link>
-  </div>
-)
+const ProjectTile = ({ data }) => {
+
+  return (
```
Comment (RanDumSocks) [inline src/components/ProjectTile.js]: Conceptually, why is a return needed here?

File: src/pages/index.js [hunk 2]
```diff
@@ -24,7 +24,7 @@ const IndexPage = ({data}) => (
             <hr className="horizontal-rule" />
                 <div className="columns is-multiline is-centered">
                   {data.allStrapiProject.edges.map(document => (
-                      <div className="column is-8-tablet is-6-desktop is-3-widescreen">
+                      <div key={document.node.id} className="column is-8-tablet is-6-desktop is-3-widescreen">
```
Comment (RanDumSocks) [inline src/pages/index.js]: I get that keys should be included in lest elements, just still not 100% sure why even after looking it up.

File: gatsby-config.js [hunk 3]
```diff
@@ -74,12 +74,6 @@ module.exports = {
         ],
       },
     },
-    {
-      resolve: 'gatsby-plugin-netlify-cms',
```
Comment (RanDumSocks) [inline gatsby-config.js]: What did this plugin do before removal, anything?

**Comment by chris-hammond** [comment]
Need to modify the ENV variables to make deploy previews work

File: src/pages/index.js (hunk 2 above)
Comment (chris-hammond) [reply src/pages/index.js]: I'm not sure either, but I know that I see console errors if it's not there...

File: gatsby-config.js (hunk 3 above)
Comment (chris-hammond) [reply gatsby-config.js]: This plugin is what made things happen when you went to /admin

**Pull request merged**